test/unit/dom-if.html

```diff
@@ -497,19 +504,132 @@
 
     });
 
-    suite('queueing race conditions', function() {
+    suite('attach/detach tests', function() {
+
+      test('remove, append domif', function(done) {
+        var domif = document.querySelector('#simple');
+        domif.if = true;
+        outerContainer.removeChild(domif);
+        setTimeout(function() {
+          var clients = outerContainer.querySelectorAll('x-client');
+          assert.equal(clients.length, 0);
+          outerContainer.appendChild(domif);
+          setTimeout(function() {
+            var clients = outerContainer.querySelectorAll('x-client');
+            assert.equal(clients[0].uid, 0);
+            assert.equal(clients[1].uid, 1);
+            assert.equal(clients[2].uid, 2);
+            assert.equal(clients[1].previousElementSibling, clients[0]);
+            assert.equal(clients[2].previousElementSibling, clients[1]);
+            assert.equal(domif.previousElementSibling, clients[2]);
```

Comment:
These 6 equals are used a number of times, with just different values. It might be possible to extract a function for this, with passing `clients` and the start number for the first 3 equals (e.g. `0` in this case and `6` for the test `move host with domif (clients persist)`). This would greatly reduce the amount of duplicate code and increase maintainability of the tests.